controller/src/main/java/org/jboss/as/controller/ModelControllerLock.java

```diff
@@ -102,12 +143,31 @@ protected synchronized boolean tryAcquire(int permit) {
         }
 
         @Override
-        protected synchronized boolean tryRelease(int permit) {
-            final Object value = permitHolder.get();
-            if (value == null) {
+        protected int tryAcquireShared(final int permit) {
+            if (permitHolder != null) {
+                return -1;
+            }
+            if (compareAndSetState(0, 1)) {
+                return 1;
+            } else {
```

Comment:
I think the best option right now is to put this back to how it was, and open a seperate issue to fix it :)